astrodendro/mass.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+from astropy import units as u
+from astropy.constants import si
+
+
+def quantity_sum(quantities):
+    """
+    In Astropy 0.3, np.sum will do the right thing for quantities, but in the mean time we need a workaround.
+    """
+    return np.sum(quantities.value) * quantities.unit
+
+
+def compute_mass(input_quantities, output_unit, spatial_scale=None):
+    """
+    Given a set of density values in Msun/pc**3 or g/cm**3 units, 
+    find the total mass in a specific set of units.
+
+    Parameters
+    ----------
+    input_quantities : `~astropy.units.quantity.Quantity` instance
+        A `~astropy.units.quantity.Quantity` instance containing an array of
+        density values to be summed.
+    output_unit : `~astropy.units.core.Unit` instance
+        The final unit to give the total mass in (should be equivalent to Msun)
+    spatial_scale : `~astropy.units.quantity.Quantity` instance
+        The pixel scale of the data
+    """
+
+    # Start off by finding the total mass in Msun
+
+    if input_quantities.unit.is_equivalent(u.Msun):
+
+        # Simply sum up the values and convert to output unit
+        total_flux = quantity_sum(input_quantities).to(u.Msun)
+
+    elif input_quantities.unit.is_equivalent(u.g / u.cm ** 3):
+
+        # Convert volume pixel to cm**3
+        if spatial_scale is None:
+            print("WARNING: spatial_scale not recognized, assumed as cm")
+            pixel_volume = u.cm ** 3          
+
+        elif spatial_scale.unit.is_equivalent(u.kpc):
+            pixel_volume = ((spatial_scale.to(u.cm)) ** 3)
+
+        elif spatial_scale.unit.is_equivalent(u.pc):        
+            pixel_volume = ((spatial_scale.to(u.cm)) ** 3)
```

Comment:
this & the next line are redundant: `u.cm`, `u.pc`, and `u.kpc` are all equivalent